tests/integration/test_s3_table_function_with_proxy/test.py

```diff
@@ -0,0 +1,94 @@
+import logging
+import os
+import time
+
+import pytest
+from helpers.cluster import ClickHouseCluster
+
+
+# Runs simple proxy resolver in python env container.
+def run_resolver(cluster):
+    container_id = cluster.get_container_id("resolver")
+    current_dir = os.path.dirname(__file__)
+    cluster.copy_file_to_container(
+        container_id,
+        os.path.join(current_dir, "proxy-resolver", "resolver.py"),
+        "resolver.py",
+    )
+    cluster.exec_in_container(container_id, ["python", "resolver.py"], detach=True)
+
+@pytest.fixture(scope="module")
+def cluster():
+    try:
+        cluster = ClickHouseCluster(__file__)
+
+        # cluster.add_instance("minio_node", with_minio=True)
+
+        # cluster.add_instance(
+        #     "proxy_list_node", main_configs=["configs/config.d/proxy_list.xml"]
+        # )
+
+        cluster.add_instance(
+            "remote_proxy_node", main_configs=["configs/config.d/proxy_remote.xml"], with_minio=True
+        )
+
+        logging.info("Starting cluster...")
+        cluster.start()
+        logging.info("Cluster started")
+
+        run_resolver(cluster)
+        logging.info("Proxy resolver started")
+
+        yield cluster
+    finally:
+        cluster.shutdown()
+
+def check_proxy_logs(cluster, proxy_instance, http_methods={"POST", "PUT", "GET"}):
+    for i in range(10):
+        logs = cluster.get_container_logs(proxy_instance)
+        # Check with retry that all possible interactions with Minio are present
+        for http_method in http_methods:
+            if logs.find(http_method + " http://minio1") >= 0:
+                return
+            time.sleep(1)
+        else:
+            assert False, "http method not found in logs"
+
+# def test_s3_with_proxy_list(cluster):
```

Comment:
This test is commented out temporarily because apparently it won't work with two minio instances. The reason for having two minio instances is that I need two CH nodes with different configs, one containing the proxy list and the other the remote one. Later one I'll need to introduce the environment one as well.

Tried using a separate node for minio, but did not work either.

Both tests pass, but not if they are run together.

I feel like there is a smarter & simpler solution to this, suggestions are appreciated.